src/main/webapp/js/ideas/ideaDetails.controller.js

```diff
@@ -5,19 +5,42 @@
         .module('app.controllers')
         .controller('detailsCtrl', detailsCtrl);
 
-    detailsCtrl.$inject = ['$scope', '$window', '$state', 'ideaDetails'];
+    detailsCtrl.$inject = ['$scope', '$window', '$state', 'ideasFactory', 'ideaDetails'];
 
-    function detailsCtrl($scope, $window, $state, ideaDetails) {
+    function detailsCtrl($scope, $window, $state, ideasFactory, ideaDetails) {
 
-        this.data = ideaDetails;
-        //$scope.data = ideaDetails;
+        $scope.idea = ideaDetails;
+        $scope.data = null;
+
+        $scope.myMap = null;
+
+        this.promises = ideasFactory.getIdeaById($scope.idea.id).then(
+                                       //success
+                                       function( value )
+                                       {
+                                        $scope.data = value;
```

Comment:
In all controllers (add, update, details) we use $scope.data in view. This is "template" :)